tests/test_utils_types.py

```diff
@@ -133,3 +134,30 @@ def test_get_metavar(self):
         which is the resource name, but in uppercase.
         """
         self.assertEqual(self.related.get_metavar(None), 'USER')
+
+
+class GeneralTests(unittest.TestCase):
+    """A set of tests that are not specific to any type.
+    """
+    def test_type_name(self):
+        """Establish that custom types are equipped with __name__ field and the
+        upstream click.Choice's issue of having no __name__ field is properly
+        dealt with.
+        """
+        names = [
+            'ad_hoc',
+            'credential',
+            'group',
+            'host',
+            'inventory',
+            'inventory_source',
+            'job',
+            'job_template',
+            'organization',
+            'project',
+            'team',
+            'user'
+        ]
+        for res_name in names:
+            res = get_resource(res_name)
+            self.assertEqual(type(res.fields), type([]))
```

Comment:
One thing I would like to know - was this failing before the fix? I know it threw the attribute error, but if that happened in a try-except block, it could still return a list with the error text as an element. But if running the tests previously showed an error, then this is fine.